packages/babel-parser/test/fixtures/typescript/class/optional-generator-method-with-modifiers/input.ts

```diff
@@ -0,0 +1,19 @@
+class C {
+  private *a?() { }
+  public *b?() { }
+  static *c?() { }
+  abstract *d?() { }
+  readonly *e?() { }
+  declare *f?() { }
+  protected *g?() { }
+}
+
+class A {
+  private *[a?.a]?() { }
+  public *[b?.b]?() { }
+  static *[c?.c]?() { }
+  abstract *[d?.d]?() { }
+  readonly *[e?.e]?() { }
+  declare *[f?.f]?() { }
+  protected *[g?.g]?() { }
+}
```

Comment:
Thank you. Can you split this test into two? one for the valid cases and the other for invalid ones.

Reply:
Done